cockpit-lib-update

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env python3
+
+# This file is part of Cockpit.
+#
+# Copyright (C) 2023 Red Hat, Inc.
+#
+# Cockpit is free software; you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# Cockpit is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+
+# Update COCKPIT_REPO_COMMIT to cockpit HEAD automatically, defaults to
+# Makefile as input optionally the full path can be provided. (For example
+# Anaconda uses ui/webui/Makefile.am).
+
+import os
+import re
+import subprocess
+import sys
+import tempfile
+from pathlib import Path
+
+sys.dont_write_bytecode = True
+
+import task
+from lib.constants import BASE_DIR
+
+GIT_URL_RE = r'COCKPIT_REPO_URL = (.*)'
+GIT_COMMIT_RE = r'COCKPIT_REPO_COMMIT = (.*)'
+
+
+def run(context, verbose=False, **kwargs):
+    cockpit_repo_url = 'https://github.com/cockpit-project/cockpit.git'
+    cockpit_repo_commit = 'HEAD'
+    makefile = context or 'Makefile'
+    makefile_path = os.path.join(BASE_DIR, makefile)
+
+    if not os.path.exists(makefile_path):
+        raise FileNotFoundError(f"Makefile '{makefile_path}' does not exists")
+
+    with open(makefile_path) as fp:
+        lines = fp.read()
+
+    m = re.search(GIT_URL_RE, lines)
+    if m:
+        cockpit_repo_url = m.group(1)
+
+    m = re.search(GIT_COMMIT_RE, lines)
+    if m:
+        cockpit_repo_commit = m.group(1)
+
+    # Figure out latest cockpit tip commit
+    with tempfile.TemporaryDirectory('cockpit-repo') as tmpdir:
+        tmpdir = Path(tmpdir)
+        clone_dir = 'cockpit'
+        subprocess.check_call(['git', 'clone', cockpit_repo_url, clone_dir], cwd=tmpdir)
+        git_describe = subprocess.check_output(['git', 'describe'], cwd=tmpdir / clone_dir).decode().strip()
+        git_head = subprocess.check_output(['git', 'rev-parse', 'HEAD'], cwd=tmpdir / clone_dir).decode().strip()
+
+    tag, commits, _ = git_describe.split('-')
+    comment = f'{git_head} # {tag} + {commits} commits'
```

Comment:
Is that safe for a "clean" tag without extra commits?

Reply:
Good question, it is!

```
[jelle@t14s][~/projects/pyalpm]%git checkout 0.9.2
Note: switching to '0.9.2'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by switching back to a branch.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -c with the switch command. Example:

  git switch -c <new-branch-name>

Or undo this operation with:

  git switch -

Turn off this advice by setting config variable advice.detachedHead to false

HEAD is now at 4428779 Release 0.9.2
[jelle@t14s][~/projects/pyalpm]%git describe --long
0.9.2-0-g4428779
```